### PR TITLE
fix(github): Ignore pending response from legacy commit check API

### LINKED ISSUE #39: craft thinks GitHub check is PENDING, GitHub shows green PASSED
https://github.com/getsentry/sentry-go/runs/423997330 reported checks PASSED but `craft` kept looping.

Could it be that the different GitHub APIs consulted were out of sync?

<details>

```
~/s/sentry-go ❯❯❯ CRAFT_LOG_LEVEL=debug craft publish 0.5.0 --skip-status-check                               master ✭
ℹ info craft 0.9.0
… debug Found environment file in the home directory: /Users/rodolfo/.craft.env
… debug Read the following variables from /Users/rodolfo/.craft.env: GITHUB_API_TOKEN,ZEUS_API_TOKEN
… debug No environment file found in the configuration directory: /Users/rodolfo/sentry/sentry-go/.craft.env
… debug Not initializing Sentry SDK
… debug Catching Ctrl-C is disabled by default. See https://github.com/getsentry/craft/issues/21
… debug Argv:
  {"_":["publish"],"no-input":false,"noInput":false,"dry-run":false,"dryRun":false,"no-merge":false,"noMerge":false,"keep-branch":false,"keepBranch":false,"keep-downloads":false,"keepDownloads":false,"no-status-check":false,"noStatusCheck":false,"skip-status-check":true,"skipStatusCheck":true,"target":"all","t":"all","$0":"/usr/local/bin/craft","NEW-VERSION":"0.5.0","newVersion":"0.5.0","new-version":"0.5.0"}
… debug Configuration file found:
  /Users/rodolfo/sentry/sentry-go/.craft.yml
… debug Parsing and validating the configuration file...
ℹ info "craft" version is compatible with the minimal version from the configuration file.
ℹ info Publishing version: "0.5.0"
… debug Fetching branch information
  release/0.5.0
… debug GET /repos/getsentry/sentry-go/branches/release/0.5.0 - 200 in 289ms
… debug Revision to publish:
  70ee6a791978e9e4b96112f30fd2cad7132ecd84
ℹ info Using "GithubStatusProvider" for status checks
… debug Fetching repository information...
… debug GET /repos/getsentry/sentry-go - 200 in 296ms
… debug Repository info received: "{
    "status": 200,
    "url": "https://api.github.com/repos/getsentry/sentry-go",
    "headers": {
        "access-control-allow-origin": "*",
        "access-control-expose-headers": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type",
        "cache-control": "private, max-age=60, s-maxage=60",
        "connection": "close",
        "content-encoding": "gzip",
        "content-security-policy": "default-src 'none'",
        "content-type": "application/json; charset=utf-8",
        "date": "Mon, 03 Feb 2020 20:26:56 GMT",
        "etag": "W/\"8ae8de0e9d2caf040054fb50123e9a61\"",
        "last-modified": "Mon, 03 Feb 2020 19:59:12 GMT",
        "referrer-policy": "origin-when-cross-origin, strict-origin-when-cross-origin",
        "server": "GitHub.com",
        "status": "200 OK",
        "strict-transport-security": "max-age=31536000; includeSubdomains; preload",
        "transfer-encoding": "chunked",
        "vary": "Accept, Authorization, Cookie, X-GitHub-OTP, Accept-Encoding",
        "x-accepted-oauth-scopes": "repo",
        "x-content-type-options": "nosniff",
        "x-frame-options": "deny",
        "x-github-media-type": "github.v3; format=json",
        "x-github-request-id": "DB30:4609:E264865:10E6546F:5E38820F",
        "x-oauth-scopes": "repo",
        "x-ratelimit-limit": "5000",
        "x-ratelimit-remaining": "4921",
        "x-ratelimit-reset": "1580763515",
        "x-xss-protection": "1; mode=block"
    },
    "data": {
        "id": 174522026,
        "node_id": "MDEwOlJlcG9zaXRvcnkxNzQ1MjIwMjY=",
        "name": "sentry-go",
        "full_name": "getsentry/sentry-go",
        "private": false,
        "owner": {
            "login": "getsentry",
            "id": 1396951,
            "node_id": "MDEyOk9yZ2FuaXphdGlvbjEzOTY5NTE=",
            "avatar_url": "https://avatars0.githubusercontent.com/u/1396951?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/getsentry",
            "html_url": "https://github.com/getsentry",
            "followers_url": "https://api.github.com/users/getsentry/followers",
            "following_url": "https://api.github.com/users/getsentry/following{/other_user}",
            "gists_url": "https://api.github.com/users/getsentry/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/getsentry/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/getsentry/subscriptions",
            "organizations_url": "https://api.github.com/users/getsentry/orgs",
            "repos_url": "https://api.github.com/users/getsentry/repos",
            "events_url": "https://api.github.com/users/getsentry/events{/privacy}",
            "received_events_url": "https://api.github.com/users/getsentry/received_events",
            "type": "Organization",
            "site_admin": false
        },
        "html_url": "https://github.com/getsentry/sentry-go",
        "description": "Official Sentry SDK for Go",
        "fork": false,
        "url": "https://api.github.com/repos/getsentry/sentry-go",
        "forks_url": "https://api.github.com/repos/getsentry/sentry-go/forks",
        "keys_url": "https://api.github.com/repos/getsentry/sentry-go/keys{/key_id}",
        "collaborators_url": "https://api.github.com/repos/getsentry/sentry-go/collaborators{/collaborator}",
        "teams_url": "https://api.github.com/repos/getsentry/sentry-go/teams",
        "hooks_url": "https://api.github.com/repos/getsentry/sentry-go/hooks",
        "issue_events_url": "https://api.github.com/repos/getsentry/sentry-go/issues/events{/number}",
        "events_url": "https://api.github.com/repos/getsentry/sentry-go/events",
        "assignees_url": "https://api.github.com/repos/getsentry/sentry-go/assignees{/user}",
        "branches_url": "https://api.github.com/repos/getsentry/sentry-go/branches{/branch}",
        "tags_url": "https://api.github.com/repos/getsentry/sentry-go/tags",
        "blobs_url": "https://api.github.com/repos/getsentry/sentry-go/git/blobs{/sha}",
        "git_tags_url": "https://api.github.com/repos/getsentry/sentry-go/git/tags{/sha}",
        "git_refs_url": "https://api.github.com/repos/getsentry/sentry-go/git/refs{/sha}",
        "trees_url": "https://api.github.com/repos/getsentry/sentry-go/git/trees{/sha}",
        "statuses_url": "https://api.github.com/repos/getsentry/sentry-go/statuses/{sha}",
        "languages_url": "https://api.github.com/repos/getsentry/sentry-go/languages",
        "stargazers_url": "https://api.github.com/repos/getsentry/sentry-go/stargazers",
        "contributors_url": "https://api.github.com/repos/getsentry/sentry-go/contributors",
        "subscribers_url": "https://api.github.com/repos/getsentry/sentry-go/subscribers",
        "subscription_url": "https://api.github.com/repos/getsentry/sentry-go/subscription",
        "commits_url": "https://api.github.com/repos/getsentry/sentry-go/commits{/sha}",
        "git_commits_url": "https://api.github.com/repos/getsentry/sentry-go/git/commits{/sha}",
        "comments_url": "https://api.github.com/repos/getsentry/sentry-go/comments{/number}",
        "issue_comment_url": "https://api.github.com/repos/getsentry/sentry-go/issues/comments{/number}",
        "contents_url": "https://api.github.com/repos/getsentry/sentry-go/contents/{+path}",
        "compare_url": "https://api.github.com/repos/getsentry/sentry-go/compare/{base}...{head}",
        "merges_url": "https://api.github.com/repos/getsentry/sentry-go/merges",
        "archive_url": "https://api.github.com/repos/getsentry/sentry-go/{archive_format}{/ref}",
        "downloads_url": "https://api.github.com/repos/getsentry/sentry-go/downloads",
        "issues_url": "https://api.github.com/repos/getsentry/sentry-go/issues{/number}",
        "pulls_url": "https://api.github.com/repos/getsentry/sentry-go/pulls{/number}",
        "milestones_url": "https://api.github.com/repos/getsentry/sentry-go/milestones{/number}",
        "notifications_url": "https://api.github.com/repos/getsentry/sentry-go/notifications{?since,all,participating}",
        "labels_url": "https://api.github.com/repos/getsentry/sentry-go/labels{/name}",
        "releases_url": "https://api.github.com/repos/getsentry/sentry-go/releases{/id}",
        "deployments_url": "https://api.github.com/repos/getsentry/sentry-go/deployments",
        "created_at": "2019-03-08T11:02:24Z",
        "updated_at": "2020-02-03T19:59:12Z",
        "pushed_at": "2020-02-03T19:59:59Z",
        "git_url": "git://github.com/getsentry/sentry-go.git",
        "ssh_url": "git@github.com:getsentry/sentry-go.git",
        "clone_url": "https://github.com/getsentry/sentry-go.git",
        "svn_url": "https://github.com/getsentry/sentry-go",
        "homepage": "",
        "size": 353,
        "stargazers_count": 199,
        "watchers_count": 199,
        "language": "Go",
        "has_issues": true,
        "has_projects": true,
        "has_downloads": true,
        "has_wiki": true,
        "has_pages": false,
        "forks_count": 33,
        "mirror_url": null,
        "archived": false,
        "disabled": false,
        "open_issues_count": 37,
        "license": {
            "key": "bsd-2-clause",
            "name": "BSD 2-Clause \"Simplified\" License",
            "spdx_id": "BSD-2-Clause",
            "url": "https://api.github.com/licenses/bsd-2-clause",
            "node_id": "MDc6TGljZW5zZTQ="
        },
        "forks": 33,
        "open_issues": 37,
        "watchers": 199,
        "default_branch": "master",
        "permissions": {
            "admin": true,
            "push": true,
            "pull": true
        },
        "temp_clone_token": "",
        "allow_squash_merge": true,
        "allow_merge_commit": true,
        "allow_rebase_merge": true,
        "delete_branch_on_merge": false,
        "organization": {
            "login": "getsentry",
            "id": 1396951,
            "node_id": "MDEyOk9yZ2FuaXphdGlvbjEzOTY5NTE=",
            "avatar_url": "https://avatars0.githubusercontent.com/u/1396951?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/getsentry",
            "html_url": "https://github.com/getsentry",
            "followers_url": "https://api.github.com/users/getsentry/followers",
            "following_url": "https://api.github.com/users/getsentry/following{/other_user}",
            "gists_url": "https://api.github.com/users/getsentry/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/getsentry/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/getsentry/subscriptions",
            "organizations_url": "https://api.github.com/users/getsentry/orgs",
            "repos_url": "https://api.github.com/users/getsentry/repos",
            "events_url": "https://api.github.com/users/getsentry/events{/privacy}",
            "received_events_url": "https://api.github.com/users/getsentry/received_events",
            "type": "Organization",
            "site_admin": false
        },
        "network_count": 33,
        "subscribers_count": 7
    }
}"
… debug Fetching combined revision status...
… debug GET /repos/getsentry/sentry-go/commits/70ee6a791978e9e4b96112f30fd2cad7132ecd84/status - 200 in 262ms
… debug Revision combined status received: "{
    "state": "pending",
    "statuses": [],
    "sha": "70ee6a791978e9e4b96112f30fd2cad7132ecd84",
    "total_count": 0,
    "repository": {
        "id": 174522026,
        "node_id": "MDEwOlJlcG9zaXRvcnkxNzQ1MjIwMjY=",
        "name": "sentry-go",
        "full_name": "getsentry/sentry-go",
        "private": false,
        "owner": {
            "login": "getsentry",
            "id": 1396951,
            "node_id": "MDEyOk9yZ2FuaXphdGlvbjEzOTY5NTE=",
            "avatar_url": "https://avatars0.githubusercontent.com/u/1396951?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/getsentry",
            "html_url": "https://github.com/getsentry",
            "followers_url": "https://api.github.com/users/getsentry/followers",
            "following_url": "https://api.github.com/users/getsentry/following{/other_user}",
            "gists_url": "https://api.github.com/users/getsentry/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/getsentry/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/getsentry/subscriptions",
            "organizations_url": "https://api.github.com/users/getsentry/orgs",
            "repos_url": "https://api.github.com/users/getsentry/repos",
            "events_url": "https://api.github.com/users/getsentry/events{/privacy}",
            "received_events_url": "https://api.github.com/users/getsentry/received_events",
            "type": "Organization",
            "site_admin": false
        },
        "html_url": "https://github.com/getsentry/sentry-go",
        "description": "Official Sentry SDK for Go",
        "fork": false,
        "url": "https://api.github.com/repos/getsentry/sentry-go",
        "forks_url": "https://api.github.com/repos/getsentry/sentry-go/forks",
        "keys_url": "https://api.github.com/repos/getsentry/sentry-go/keys{/key_id}",
        "collaborators_url": "https://api.github.com/repos/getsentry/sentry-go/collaborators{/collaborator}",
        "teams_url": "https://api.github.com/repos/getsentry/sentry-go/teams",
        "hooks_url": "https://api.github.com/repos/getsentry/sentry-go/hooks",
        "issue_events_url": "https://api.github.com/repos/getsentry/sentry-go/issues/events{/number}",
        "events_url": "https://api.github.com/repos/getsentry/sentry-go/events",
        "assignees_url": "https://api.github.com/repos/getsentry/sentry-go/assignees{/user}",
        "branches_url": "https://api.github.com/repos/getsentry/sentry-go/branches{/branch}",
        "tags_url": "https://api.github.com/repos/getsentry/sentry-go/tags",
        "blobs_url": "https://api.github.com/repos/getsentry/sentry-go/git/blobs{/sha}",
        "git_tags_url": "https://api.github.com/repos/getsentry/sentry-go/git/tags{/sha}",
        "git_refs_url": "https://api.github.com/repos/getsentry/sentry-go/git/refs{/sha}",
        "trees_url": "https://api.github.com/repos/getsentry/sentry-go/git/trees{/sha}",
        "statuses_url": "https://api.github.com/repos/getsentry/sentry-go/statuses/{sha}",
        "languages_url": "https://api.github.com/repos/getsentry/sentry-go/languages",
        "stargazers_url": "https://api.github.com/repos/getsentry/sentry-go/stargazers",
        "contributors_url": "https://api.github.com/repos/getsentry/sentry-go/contributors",
        "subscribers_url": "https://api.github.com/repos/getsentry/sentry-go/subscribers",
        "subscription_url": "https://api.github.com/repos/getsentry/sentry-go/subscription",
        "commits_url": "https://api.github.com/repos/getsentry/sentry-go/commits{/sha}",
        "git_commits_url": "https://api.github.com/repos/getsentry/sentry-go/git/commits{/sha}",
        "comments_url": "https://api.github.com/repos/getsentry/sentry-go/comments{/number}",
        "issue_comment_url": "https://api.github.com/repos/getsentry/sentry-go/issues/comments{/number}",
        "contents_url": "https://api.github.com/repos/getsentry/sentry-go/contents/{+path}",
        "compare_url": "https://api.github.com/repos/getsentry/sentry-go/compare/{base}...{head}",
        "merges_url": "https://api.github.com/repos/getsentry/sentry-go/merges",
        "archive_url": "https://api.github.com/repos/getsentry/sentry-go/{archive_format}{/ref}",
        "downloads_url": "https://api.github.com/repos/getsentry/sentry-go/downloads",
        "issues_url": "https://api.github.com/repos/getsentry/sentry-go/issues{/number}",
        "pulls_url": "https://api.github.com/repos/getsentry/sentry-go/pulls{/number}",
        "milestones_url": "https://api.github.com/repos/getsentry/sentry-go/milestones{/number}",
        "notifications_url": "https://api.github.com/repos/getsentry/sentry-go/notifications{?since,all,participating}",
        "labels_url": "https://api.github.com/repos/getsentry/sentry-go/labels{/name}",
        "releases_url": "https://api.github.com/repos/getsentry/sentry-go/releases{/id}",
        "deployments_url": "https://api.github.com/repos/getsentry/sentry-go/deployments"
    },
    "commit_url": "https://api.github.com/repos/getsentry/sentry-go/commits/70ee6a791978e9e4b96112f30fd2cad7132ecd84",
    "url": "https://api.github.com/repos/getsentry/sentry-go/commits/70ee6a791978e9e4b96112f30fd2cad7132ecd84/status"
}"
… debug Fetching Checks API status...
… debug GET /repos/getsentry/sentry-go/commits/70ee6a791978e9e4b96112f30fd2cad7132ecd84/check-runs - 200 in 304ms
… debug Revision checks received: "{
    "total_count": 1,
    "check_runs": [
        {
            "id": 423939391,
            "node_id": "MDg6Q2hlY2tSdW40MjM5MzkzOTE=",
            "head_sha": "70ee6a791978e9e4b96112f30fd2cad7132ecd84",
            "external_id": "147270711",
            "url": "https://api.github.com/repos/getsentry/sentry-go/check-runs/423939391",
            "html_url": "https://github.com/getsentry/sentry-go/runs/423939391",
            "details_url": "https://travis-ci.com/getsentry/sentry-go/builds/147270711",
            "status": "completed",
            "conclusion": "success",
            "started_at": "2020-02-03T20:00:20Z",
            "completed_at": "2020-02-03T20:03:03Z",
            "output": {
                "title": "Build Passed",
                "summary": "<a href='https://travis-ci.com/getsentry/sentry-go/builds/147270711'><img src='https://travis-ci.com/images/stroke-icons/icon-passed.png' height='11'> The build</a> **passed**.",
                "text": "This is a normal build for the release\\/0.5.0 branch. You should be able to reproduce it by checking out the branch locally.\n\n## Jobs and Stages\nThis build has **nine jobs**, running in parallel.\n\n<table>\n<thead>\n  <tr>\n    <th>Job</th>\n    <th>Go</th>\n    <th>ENV</th>\n    <th>State</th>\n    <th>Notes</th>\n  </tr>\n</thead>\n<tbody>\n  <tr>\n    <td><a href='https://travis-ci.com/getsentry/sentry-go/jobs/283142896'><img src='https://travis-ci.com/images/stroke-icons/icon-passed.png' height='11'> 400.1</a></td>\n    <td>1.11.x</td>\n    <td>GO111MODULE=on GOFLAGS=-mod=readonly</td>\n    <td>passed</td>\n    <td></td>\n  </tr>\n  <tr>\n    <td><a href='https://travis-ci.com/getsentry/sentry-go/jobs/283142897'><img src='https://travis-ci.com/images/stroke-icons/icon-passed.png' height='11'> 400.2</a></td>\n    <td>1.11.x</td>\n    <td>GO111MODULE=off</td>\n    <td>passed</td>\n    <td></td>\n  </tr>\n  <tr>\n    <td><a href='https://travis-ci.com/getsentry/sentry-go/jobs/283142898'><img src='https://travis-ci.com/images/stroke-icons/icon-passed.png' height='11'> 400.3</a></td>\n    <td>1.12.x</td>\n    <td>GO111MODULE=on GOFLAGS=-mod=readonly</td>\n    <td>passed</td>\n    <td></td>\n  </tr>\n  <tr>\n    <td><a href='https://travis-ci.com/getsentry/sentry-go/jobs/283142899'><img src='https://travis-ci.com/images/stroke-icons/icon-passed.png' height='11'> 400.4</a></td>\n    <td>1.12.x</td>\n    <td>GO111MODULE=off</td>\n    <td>passed</td>\n    <td></td>\n  </tr>\n  <tr>\n    <td><a href='https://travis-ci.com/getsentry/sentry-go/jobs/283142900'><img src='https://travis-ci.com/images/stroke-icons/icon-passed.png' height='11'> 400.5</a></td>\n    <td>1.13.x</td>\n    <td>GO111MODULE=on GOFLAGS=-mod=readonly</td>\n    <td>passed</td>\n    <td></td>\n  </tr>\n  <tr>\n    <td><a href='https://travis-ci.com/getsentry/sentry-go/jobs/283142901'><img src='https://travis-ci.com/images/stroke-icons/icon-passed.png' height='11'> 400.6</a></td>\n    <td>1.13.x</td>\n    <td>GO111MODULE=off</td>\n    <td>passed</td>\n    <td></td>\n  </tr>\n  <tr>\n    <td><a href='https://travis-ci.com/getsentry/sentry-go/jobs/283142902'><img src='https://travis-ci.com/images/stroke-icons/icon-passed.png' height='11'> 400.7 Module support outside of GOPATH</a></td>\n    <td>1.13.x</td>\n    <td>GO111MODULE=on GOFLAGS=-mod=readonly</td>\n    <td>passed</td>\n    <td></td>\n  </tr>\n  <tr>\n    <td><a href='https://travis-ci.com/getsentry/sentry-go/jobs/283142903'><img src='https://travis-ci.com/images/stroke-icons/icon-passed.png' height='11'> 400.8</a></td>\n    <td>master</td>\n    <td>GO111MODULE=on GOFLAGS=-mod=readonly</td>\n    <td>passed</td>\n    <td>This job is <a href='https://docs.travis-ci.com/user/customizing-the-build#Rows-that-are-Allowed-to-Fail'>allowed to fail</a>.</td>\n  </tr>\n  <tr>\n    <td><a href='https://travis-ci.com/getsentry/sentry-go/jobs/283142904'><img src='https://travis-ci.com/images/stroke-icons/icon-passed.png' height='11'> 400.9</a></td>\n    <td>master</td>\n    <td>GO111MODULE=off</td>\n    <td>passed</td>\n    <td>This job is <a href='https://docs.travis-ci.com/user/customizing-the-build#Rows-that-are-Allowed-to-Fail'>allowed to fail</a>.</td>\n  </tr>\n</tbody>\n</table>\n\n## Build Configuration\n\nBuild Option     | Setting\n-----------------|--------------\nLanguage         | Go\nOperating System | Linux (Xenial)\nGo Versions      | 1.11.x, 1.12.x, 1.13.x, master\n\n<details>\n<summary>Build Configuration</summary>\n<pre lang='yaml'>\n{\n  \"go\": [\n    \"1.11.x\",\n    \"1.12.x\",\n    \"1.13.x\",\n    \"master\"\n  ],\n  \"os\": \"linux\",\n  \"env\": [\n    \"GO111MODULE=on GOFLAGS=-mod=readonly\",\n    \"GO111MODULE=off\"\n  ],\n  \"dist\": \"xenial\",\n  \"jobs\": {\n    \"include\": [\n      {\n        \"go\": \"1.13.x\",\n        \"name\": \"Module support outside of GOPATH\",\n        \"script\": \"go test && go test -race\",\n        \"before_script\": \"mv $GOPATH/src/github.com/getsentry/sentry-go ~/sentry-go && cd ~/sentry-go && export GOPATH= && go env GOPATH\"\n      }\n    ],\n    \"fast_finish\": true,\n    \"allow_failures\": [\n      {\n        \"go\": \"master\"\n      }\n    ]\n  },\n  \"group\": \"stable\",\n  \"script\": [\n    \"golangci-lint run --new-from-rev=$(git merge-base origin/master HEAD)\",\n    \"go build\",\n    \"go test\",\n    \"go test -race\"\n  ],\n  \".result\": \"configured\",\n  \"language\": \"go\",\n  \"notifications\": {\n    \"webhooks\": {\n      \"urls\": [\n        \"https://zeus.ci/hooks/befe9810-9285-11e9-b01a-0a580a281808/public/provider/travis/webhook\"\n      ],\n      \"on_error\": \"always\",\n      \"on_start\": \"always\",\n      \"on_cancel\": \"always\",\n      \"on_failure\": \"always\",\n      \"on_success\": \"always\"\n    }\n  },\n  \"before_install\": [\n    \"curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/v1.19.1/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.19.1\",\n    \"git fetch origin master:remotes/origin/master\"\n  ]\n}\n</pre>\n</details>",
                "annotations_count": 0,
                "annotations_url": "https://api.github.com/repos/getsentry/sentry-go/check-runs/423939391/annotations"
            },
            "name": "Travis CI - Branch",
            "check_suite": {
                "id": 435789300
            },
            "app": {
                "id": 67,
                "slug": "travis-ci",
                "node_id": "MDM6QXBwNjc=",
                "owner": {
                    "login": "travis-ci",
                    "id": 639823,
                    "node_id": "MDEyOk9yZ2FuaXphdGlvbjYzOTgyMw==",
                    "avatar_url": "https://avatars2.githubusercontent.com/u/639823?v=4",
                    "gravatar_id": "",
                    "url": "https://api.github.com/users/travis-ci",
                    "html_url": "https://github.com/travis-ci",
                    "followers_url": "https://api.github.com/users/travis-ci/followers",
                    "following_url": "https://api.github.com/users/travis-ci/following{/other_user}",
                    "gists_url": "https://api.github.com/users/travis-ci/gists{/gist_id}",
                    "starred_url": "https://api.github.com/users/travis-ci/starred{/owner}{/repo}",
                    "subscriptions_url": "https://api.github.com/users/travis-ci/subscriptions",
                    "organizations_url": "https://api.github.com/users/travis-ci/orgs",
                    "repos_url": "https://api.github.com/users/travis-ci/repos",
                    "events_url": "https://api.github.com/users/travis-ci/events{/privacy}",
                    "received_events_url": "https://api.github.com/users/travis-ci/received_events",
                    "type": "Organization",
                    "site_admin": false
                },
                "name": "Travis CI",
                "description": "Test and deploy with confidence. Trusted by over 800,000 users, Travis CI is the leading hosted continuous integration system.\r\n\r\nSupporting over 30 different languages, including Ruby, Mac/iOS, and Docker, Travis CI is built for everyone.\r\n\r\nFree for open source, and with a 100 build trial for private projects, getting setup takes just 2 minutes.",
                "external_url": "https://travis-ci.com",
                "html_url": "https://github.com/apps/travis-ci",
                "created_at": "2016-06-21T16:22:21Z",
                "updated_at": "2018-09-14T20:36:16Z",
                "permissions": {
                    "checks": "write",
                    "contents": "read",
                    "deployments": "write",
                    "members": "read",
                    "metadata": "read",
                    "pull_requests": "read",
                    "repository_hooks": "write",
                    "statuses": "write"
                },
                "events": [
                    "check_run",
                    "check_suite",
                    "create",
                    "delete",
                    "member",
                    "pull_request",
                    "push",
                    "repository"
                ]
            },
            "pull_requests": []
        }
    ]
}"
ℹ info No config provided for Github status provider, calculating the combined status...
ℹ info pending
ℹ info Revision 70ee6a791978e9e4b96112f30fd2cad7132ecd84 has been found.
[==  ] [2/3/2020, 9:26:56 PM] CI builds are still in progress, sleeping for 30 seconds...^C
```

</details>

### PR DESCRIPTION
**Pull request by tonyo**
Fixes #39 

+ minor refactoring and some docs.

### CODE REVIEW

File: src/status_providers/github.ts [hunk 1]
```diff
@@ -43,15 +41,9 @@ export class GithubStatusProvider extends BaseStatusProvider {
...
-    // Examples: Appveyor
+    const revisionStatus = await this.getCommitApiStatus(revision);
     // 2. Check runs API
-    // https://developer.github.com/v3/checks/runs/#list-check-runs-for-a-specific-ref
-    // Examples: Travis CI, Azure Pipelines
-
-    const [revisionStatus, revisionChecks] = await this.getAllStatuses(
-      revision
-    );
+    const revisionChecks = await this.getRevisionChecks(revision);
```
Comment (BYK) [inline src/status_providers/github.ts:46]:
```suggestion
const [revisionStatus, revisionChecks] = await Promise.all([
    this.getCommitApiStatus(revision),
    this.getRevisionChecks(revision),
]);
```

File: src/status_providers/github.ts [hunk 2]
```diff
@@ -79,10 +71,34 @@ export class GithubStatusProvider extends BaseStatusProvider {
...
+        revisionStatus.state === 'pending'
+      ) {
+        // Edge case, this is what GitHub returns when there are no registered legacy checks.
+        logger.debug('No legacy checks detected, checking for runs...');
+        if (revisionChecks.total_count > 0) {
+          logger.debug('Check runs exist, continuing...');
+          commitApiStatusResult = CommitStatus.SUCCESS;
+        } else {
+          logger.warn('No valid build contexts detected, did any checks run?');
+          return CommitStatus.FAILURE;
```
Comment (BYK) [inline src/status_providers/github.ts:87]: Wouldn't it be better to do
```js
commitApiStatusResult = CommitStatus.NOT_FOUND
```

and let the rest of the flow take care of that (around line 106 below)?

**Comment by BYK** [comment]
Should probably add an entry to the changlog now

File: src/status_providers/github.ts (hunk 2 above)
Comment (tonyo) [reply src/status_providers/github.ts:87]: Yes, makes sense 👍

**Pull request merged**